diff --git a/data/platform.toml b/data/platform.toml
index ddab3e5..404d8fa 100644
--- a/data/platform.toml
+++ b/data/platform.toml
@@ -1,27 +1,27 @@
 [factory_reset.all]
-script  = "/usr/bin/steamos-reset-tool"
+script  = "@stubs@/bin/steamos-reset-tool"
 script_args = ["factory-reset", "--reset-all"]
 
 [factory_reset.os]
-script  = "/usr/bin/steamos-reset-tool"
+script  = "@stubs@/bin/steamos-reset-tool"
 script_args = ["factory-reset", "--reset-os"]
 
 [factory_reset.user]
-script  = "/usr/bin/steamos-reset-tool"
+script  = "@stubs@/bin/steamos-reset-tool"
 script_args = ["factory-reset", "--reset-user-data"]
 
 [update_bios]
-script = "/usr/bin/jupiter-biosupdate"
+script = "@steamDeckFirmware@/bin/jupiter-biosupdate"
 script_args = ["--auto"]
 
 [update_dock]
-script = "/usr/lib/jupiter-dock-updater/jupiter-dock-updater.sh"
+script = "@jupiterDockUpdaterBin@/jupiter-dock-updater/jupiter-dock-updater.sh"
 
 [storage.trim_devices]
-script = "/usr/lib/hwsupport/trim-devices.sh"
+script = "@hwsupport@/lib/hwsupport/trim-devices.sh"
 
 [storage.format_device]
-script = "/usr/lib/hwsupport/format-device.sh"
+script = "@hwsupport@/lib/hwsupport/format-device.sh"
 label_flag = "--label"
 device_flag = "--device"
 no_validate_flag = "--skip-validation"
diff --git a/data/system/com.steampowered.SteamOSManager1.service b/data/system/com.steampowered.SteamOSManager1.service
index 5fcbc63..1dd1cc6 100644
--- a/data/system/com.steampowered.SteamOSManager1.service
+++ b/data/system/com.steampowered.SteamOSManager1.service
@@ -1,5 +1,5 @@
 [D-BUS Service]
 Name=com.steampowered.SteamOSManager1
-Exec=/bin/false
+Exec=@coreutils@/bin/false
 User=root
 SystemdService=steamos-manager.service
diff --git a/data/system/steamos-manager.service b/data/system/steamos-manager.service
index e49ea15..845d3de 100644
--- a/data/system/steamos-manager.service
+++ b/data/system/steamos-manager.service
@@ -9,7 +9,7 @@ StartLimitBurst=5
 Type=notify-reload
 BusName=com.steampowered.SteamOSManager1
 Environment=RUST_LOG=info
-ExecStart=/usr/lib/steamos-manager -r
+ExecStart=@out@/lib/steamos-manager -r
 Restart=on-failure
 RestartSec=1
 RestartMaxDelaySec=10
diff --git a/data/user/com.steampowered.SteamOSManager1.service b/data/user/com.steampowered.SteamOSManager1.service
index ac590d7..96340ab 100644
--- a/data/user/com.steampowered.SteamOSManager1.service
+++ b/data/user/com.steampowered.SteamOSManager1.service
@@ -1,5 +1,5 @@
 [D-BUS Service]
 Name=com.steampowered.SteamOSManager1
-Exec=/bin/false
+Exec=@coreutils@/bin/false
 SystemdService=steamos-manager.service
 
diff --git a/data/user/steamos-manager-session-cleanup.service b/data/user/steamos-manager-session-cleanup.service
index ba2b6d0..c847907 100644
--- a/data/user/steamos-manager-session-cleanup.service
+++ b/data/user/steamos-manager-session-cleanup.service
@@ -5,7 +5,7 @@ After=default.target steamos-manager.service
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/steamosctl clean-temporary-sessions
+ExecStart=@out@/bin/steamosctl clean-temporary-sessions
 
 [Install]
 WantedBy=graphical-session.target
diff --git a/data/user/steamos-manager.service b/data/user/steamos-manager.service
index c6dc9fe..96368ad 100644
--- a/data/user/steamos-manager.service
+++ b/data/user/steamos-manager.service
@@ -7,7 +7,7 @@ StartLimitBurst=5
 Type=notify-reload
 BusName=com.steampowered.SteamOSManager1
 Environment=RUST_LOG=info
-ExecStart=/usr/lib/steamos-manager
+ExecStart=@out@/lib/steamos-manager
 Restart=on-failure
 RestartSec=1
 RestartMaxDelaySec=10
diff --git a/steamos-manager-macros/src/lib.rs b/steamos-manager-macros/src/lib.rs
index 0353423..2fa3ac3 100644
--- a/steamos-manager-macros/src/lib.rs
+++ b/steamos-manager-macros/src/lib.rs
@@ -860,7 +860,7 @@ pub fn remote_manager(input: TokenStream) -> TokenStream {
                 let builder = ConfigBuilder::<AsyncState>::default();
                 let builder = read_config_directory(
                     builder,
-                    path("/usr/share/steamos-manager/remotes.d"),
+                    path("@out@/share/steamos-manager/remotes.d"),
                     FileFormat::Toml.file_extensions(),
                     FileFormat::Toml,
                 )
diff --git a/steamos-manager/src/audio.rs b/steamos-manager/src/audio.rs
index 7b14d10..1364f7d 100644
--- a/steamos-manager/src/audio.rs
+++ b/steamos-manager/src/audio.rs
@@ -49,9 +49,9 @@ impl AudioManager {
 
     #[cfg(not(test))]
     pub async fn is_supported() -> Result<bool> {
-        if try_exists("/usr/bin/wpctl").await? {
+        if try_exists("@wireplumber@/bin/wpctl").await? {
             // wpctl exists, now check if node.features.audio.mono is patched in also
-            let output = script_output("pw-dump", &["sm-settings"]).await?;
+            let output = script_output("@pipewire@/bin/pw-dump", &["sm-settings"]).await?;
             let object: Value = serde_json::from_str(&output)?;
             // Now check for our key
             if let Some(array) = object.as_array() {
@@ -90,7 +90,7 @@ impl AudioManager {
                 Mode::Stereo => "false",
             };
 
-            run_script("/usr/bin/wpctl", &["settings", MONO_KEY, mono]).await?;
+            run_script("@wireplumber@/bin/wpctl", &["settings", MONO_KEY, mono]).await?;
         }
 
         self.mode = Some(mode);
@@ -99,7 +99,7 @@ impl AudioManager {
 
     #[cfg(not(test))]
     async fn load_values(&mut self) -> Result<()> {
-        let output = script_output("/usr/bin/wpctl", &["settings", MONO_KEY]).await?;
+        let output = script_output("@wireplumber@/bin/wpctl", &["settings", MONO_KEY]).await?;
 
         match output.trim() {
             "Value: true" => self.mode = Some(Mode::Mono),
diff --git a/steamos-manager/src/daemon/root.rs b/steamos-manager/src/daemon/root.rs
index 5e7a4f4..d443eb9 100644
--- a/steamos-manager/src/daemon/root.rs
+++ b/steamos-manager/src/daemon/root.rs
@@ -108,7 +108,7 @@ impl DaemonContext for RootContext {
     }
 
     fn system_config_path(&self) -> Result<PathBuf> {
-        Ok(path("/usr/share/steamos-manager/system.d"))
+        Ok(path("@out@/share/steamos-manager/system.d"))
     }
 
     fn state(&self) -> &RootState {
diff --git a/steamos-manager/src/daemon/user.rs b/steamos-manager/src/daemon/user.rs
index 6b3a217..cf33105 100644
--- a/steamos-manager/src/daemon/user.rs
+++ b/steamos-manager/src/daemon/user.rs
@@ -81,7 +81,7 @@ impl DaemonContext for UserContext {
     }
 
     fn system_config_path(&self) -> Result<PathBuf> {
-        Ok(path("/usr/share/steamos-manager/user.d"))
+        Ok(path("@out@/share/steamos-manager/user.d"))
     }
 
     fn state(&self) -> &UserState {
diff --git a/steamos-manager/src/hardware.rs b/steamos-manager/src/hardware.rs
index 0fea30f..710f628 100644
--- a/steamos-manager/src/hardware.rs
+++ b/steamos-manager/src/hardware.rs
@@ -34,7 +34,7 @@ const SYS_VENDOR_PATH: &str = "/sys/class/dmi/id/sys_vendor";
 const BOARD_NAME_PATH: &str = "/sys/class/dmi/id/board_name";
 const PRODUCT_NAME_PATH: &str = "/sys/class/dmi/id/product_name";
 #[cfg(not(test))]
-const DEVICE_CONFIG_PATH: &str = "/usr/share/steamos-manager/devices";
+const DEVICE_CONFIG_PATH: &str = "@out@/share/steamos-manager/devices";
 #[cfg(test)]
 const DEVICE_CONFIG_PATH: &str = "../data/devices";
 
diff --git a/steamos-manager/src/manager/root.rs b/steamos-manager/src/manager/root.rs
index d58478a..3473360 100644
--- a/steamos-manager/src/manager/root.rs
+++ b/steamos-manager/src/manager/root.rs
@@ -183,7 +183,7 @@ impl SteamOSManager {
         };
         for index in indices {
             let result = script_output(
-                "/usr/bin/steamos-polkit-helpers/jupiter-get-als-gain",
+                "@polkitHelpers@/bin/steamos-polkit-helpers/jupiter-get-als-gain",
                 &["-s", index],
             )
             .await;
diff --git a/steamos-manager/src/manager/user.rs b/steamos-manager/src/manager/user.rs
index 23b6712..cea69b1 100644
--- a/steamos-manager/src/manager/user.rs
+++ b/steamos-manager/src/manager/user.rs
@@ -1486,10 +1486,6 @@ impl Service for ScreenReaderSetupService {
     const NAME: &'static str = "screenreader-setup";
 
     async fn run(&mut self) -> Result<()> {
-        if !try_exists(path("/usr/bin/orca")).await? {
-            return Ok(());
-        }
-
         let object_server = self.session.object_server();
         let orca_manager = OrcaManager::new(&self.session).await?;
         let screen_reader = Arc::new(Mutex::new(orca_manager));
diff --git a/steamos-manager/src/platform.rs b/steamos-manager/src/platform.rs
index 48b385d..d8560ed 100644
--- a/steamos-manager/src/platform.rs
+++ b/steamos-manager/src/platform.rs
@@ -174,7 +174,7 @@ impl FormatDeviceConfig {
 impl PlatformConfig {
     #[cfg(not(test))]
     async fn load() -> Result<Option<PlatformConfig>> {
-        let path = "/usr/share/steamos-manager/platform.toml";
+        let path = "@out@/share/steamos-manager/platform.toml";
         let config = read_to_string(path)
             .await
             .with_context(|| format!("Failed to read {path}"))?;
diff --git a/steamos-manager/src/power.rs b/steamos-manager/src/power.rs
index 291d5a9..96979e7 100644
--- a/steamos-manager/src/power.rs
+++ b/steamos-manager/src/power.rs
@@ -351,10 +351,7 @@ impl<'dbus> CpuSchedulerManager<'dbus> {
 
     #[cfg(not(test))]
     pub async fn is_supported() -> Result<bool> {
-        if try_exists(path("/usr/bin/scx_lavd")).await? {
-            return Ok(true);
-        }
-        Ok(false)
+        Ok(true)
     }
 
     pub(crate) async fn get_available_cpu_schedulers(&self) -> Result<Vec<CpuScheduler>> {
diff --git a/steamos-manager/src/session.rs b/steamos-manager/src/session.rs
index 73f9b2d..8fac1be 100644
--- a/steamos-manager/src/session.rs
+++ b/steamos-manager/src/session.rs
@@ -91,7 +91,7 @@ pub(crate) enum SessionManagerMessage {
 }
 
 pub(crate) async fn is_session_managed() -> Result<bool> {
-    Ok(try_exists(path(CONFIG_PREFIX).join(SESSION_CHECK_PATH)).await?)
+    Ok(true)
 }
 
 #[cfg(test)]
diff --git a/steamos-manager/src/wifi.rs b/steamos-manager/src/wifi.rs
index 02c6679..509a409 100644
--- a/steamos-manager/src/wifi.rs
+++ b/steamos-manager/src/wifi.rs
@@ -32,19 +32,18 @@ use crate::{path, read_config_directory};
 
 const OVERRIDE_CONTENTS: &str = "[Service]
 ExecStart=
-ExecStart=/usr/lib/iwd/iwd -d
+ExecStart=@iwd@/libexec/iwd -d
 ";
 const OVERRIDE_FOLDER: &str = "/etc/systemd/system/iwd.service.d";
 const OVERRIDE_PATH: &str = "/etc/systemd/system/iwd.service.d/99-valve-override.conf";
 
 // Only use one path for output for now. If needed we can add a timestamp later
 // to have multiple files, etc.
-const TRACE_CMD_PATH: &str = "/usr/bin/trace-cmd";
+const TRACE_CMD_PATH: &str = "@traceCmd@/bin/trace-cmd";
 
 const MIN_BUFFER_SIZE: u32 = 100;
 
 const WIFI_BACKEND_PATHS: &[&str] = &[
-    "/usr/lib/NetworkManager/conf.d",
     "/etc/NetworkManager/conf.d",
 ];
 
@@ -228,11 +227,11 @@ pub(crate) async fn get_wifi_backend() -> Result<WifiBackend> {
 }
 
 pub(crate) async fn set_wifi_backend(backend: WifiBackend) -> Result<()> {
-    run_script("/usr/bin/steamos-wifi-set-backend", &[backend.to_string()]).await
+    run_script("@stubs@/bin/steamos-wifi-set-backend", &[backend.to_string()]).await
 }
 
 pub(crate) async fn list_wifi_interfaces() -> Result<Vec<String>> {
-    let output = script_output("/usr/bin/iw", &["dev"]).await?;
+    let output = script_output("@iw@/bin/iw", &["dev"]).await?;
     Ok(output
         .lines()
         .filter_map(|line| match line.trim().split_once(' ') {
@@ -246,7 +245,7 @@ pub(crate) async fn get_wifi_power_management_state() -> Result<WifiPowerManagem
     let mut found_any = false;
     for iface in list_wifi_interfaces().await? {
         let output =
-            script_output("/usr/bin/iw", &["dev", iface.as_str(), "get", "power_save"]).await?;
+            script_output("@iw@/bin/iw", &["dev", iface.as_str(), "get", "power_save"]).await?;
         for line in output.lines() {
             match line.trim() {
                 "Power save: on" => return Ok(WifiPowerManagement::Enabled),
@@ -267,7 +266,7 @@ pub(crate) async fn set_wifi_power_management_state(state: WifiPowerManagement)
 
     for iface in list_wifi_interfaces().await? {
         run_script(
-            "/usr/bin/iw",
+            "@iw@/bin/iw",
             &["dev", iface.as_str(), "set", "power_save", state],
         )
         .await
